### PR TITLE
custom_fields: enabling visibility on deposit form

### PR DESCRIPTION
**Pull request by 0einstein0**
* Closes Issue [#632](https://github.com/zenodo/rdm-project/issues/632)
*closes https://github.com/zenodo/rdm-project/issues/649

Screenshot 2024-01-22 at 10 50 52
Screenshot 2024-01-22 at 10 51 10

### CODE REVIEW

File: site/zenodo_rdm/custom_fields/domain_fields.py [hunk 1]
```diff
@@ -167,381 +90,380 @@ def field(self):
...
             ),
         ),
+    ],
+}
+
+AUDIOVIS_FIELDS_UI = {
+    "section": _("Audiovisual core"),
+    "icon": "camera",
+    "hide_from_upload_form": False,
+    "compose_fields": True,  # marks the group as dynamic
```
Comment (zzacharo) [inline site/zenodo_rdm/custom_fields/domain_fields.py:428]: We need a better naming for this :) Not sure what yet.....

Comment (zzacharo) [reply site/zenodo_rdm/custom_fields/domain_fields.py:428]: what about be more explicit and say something like 
- `show_all_fields_by_default`
- `show_fields_on_demand`
- `search_fields` (kind of dropdown paradigm of show options vs search for options)

File: site/zenodo_rdm/custom_fields/__init__.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+]
+
+
+# hide meeting section from Additional details in landing page
+MEETING_CUSTOM_FIELDS_UI["hide_from_landing_page"] = True
+
+# Custom fields UI components
+CUSTOM_FIELDS_UI = [
+    # codemeta
+    # CODEMETA_CUSTOM_FIELDS_UI,  # TODO enable when fixed (see https://github.com/zenodo/rdm-project/issues/217)
```
Comment (zzacharo) [inline site/zenodo_rdm/custom_fields/__init__.py:87]: does it make sense to check with @0einstein0 if we enable it to check if it's an easy fix?

File: site/zenodo_rdm/custom_fields/__init__.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    **THESIS_NAMESPACE,
+}
+
+
+CUSTOM_FIELDS = [
+    *BIODIV_FIELDS,
+    *AUDIOVIS_FIELDS,
+    # codemeta,
+    *CODEMETA_CUSTOM_FIELDS,
+    # TODO enable when fixed (see https://github.com/zenodo/rdm-project/issues/217)
```
Comment (zzacharo) [inline site/zenodo_rdm/custom_fields/__init__.py:67]: leftover?

File: site/zenodo_rdm/custom_fields/__init__.py (hunk 2 above)
Comment (kpsherva) [reply site/zenodo_rdm/custom_fields/__init__.py:87]: it was not meant as a part of this sprint if I understood correctly

**Review by zzacharo — APPROVED** [review]
Discussed with @kpsherva IRL that the naming comment will be handled after the first merge of the chain. If so, then we can go ahead and merge this!

**Pull request merged**